app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,245 @@ def self.display_methods
+      :url  => "/security_group/edit/#{@security_group.id}")
+  end
+
+  def get_checked_security_group_id(params)
+    if params[:id]
+      checked_security_group_id = params[:id]
+    else
+      checked_security_groups = find_checked_items
+      checked_security_group_id = checked_security_groups[0] if checked_security_groups.length == 1
+    end
+    checked_security_group_id
+  end
+
+  def new
+    assert_privileges("security_group_new")
+    @security_group = SecurityGroup.new
+    @in_a_form = true
+
+    @ems_choices = {}
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
+      @ems_choices[ems.name] = ems.id
+    end
+
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+
+    drop_breadcrumb(:name => _("Add New Security Group"), :url => "/security_group/new")
+  end
+
+  def update
+    assert_privileges("security_group_edit")
+    @security_group = find_by_id_filtered(SecurityGroup, params[:id])
+    options = form_params
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of Security Group \"%{name}\" was cancelled by the user") % {
+        :name => @security_group.name
+      })
+
+    when "save"
+      task_id = @security_group.update_security_group_queue(session[:userid], options)
+
+      add_flash(_("Security Group update failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+
+      if @flash_array
+        javascript_flash(:spinner_off => true)
+      else
+        initiate_wait_for_task(:task_id => task_id, :action => "update_finished")
+      end
+    end
+  end
+
+  def update_finished
+    task_id = session[:async][:params][:task_id]
+    security_group_id = session[:async][:params][:id]
+    security_group_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("Security Group \"%{name}\" updated") % {
+        :model => ui_lookup(:table => 'security_group'),
+        :name  => security_group_name
+      })
+    else
+      add_flash(_("Unable to update Security Group \"%{name}\": %{details}") % {
+        :model   => ui_lookup(:table => 'security_group'),
+        :name    => security_group_name,
+        :details => task.message
+      }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show", :id => security_group_id
+  end
+
+  private
+
+  def form_params
+    options = {}
+    options[:name] = params[:name] if params[:name]
+    options[:description] = params[:description] if params[:description]
+    options[:ems_id] = params[:ems_id] if params[:ems_id]
+    options[:cloud_tenant_id] = params[:cloud_tenant_id] if params[:cloud_tenant_id]
```

Comment:
This needs to be something like:

options[:cloud_tenant] = find_by_id_filtered(CloudTenant, params[:cloud_tenant_id]) if params[:cloud_tenant_id]

Because the model expects options to include :cloud_tenant, and not :cloud_tenant_id (I'm currently fixing this in other places)